### DISCUSSION TITLE
how to get image tags by Docker Registry Protocol v2

### DISCUSSION THREAD

**Discussion by inajob** [quay/quay#1806] [Q&A] [answered]
I found image tags by Docker Registry Protocol v2 is different from actual tags.

- https://quay.io/v2/ansible/creator-ee/tags/list
    - it returns only until v0.9.2, but this image has until v0.14.1.

Quay has no more support this protocol?

**Comment by flavianmissi** [comment, marked as answer]
Hi @inajob, that endpoint[1] uses pagination. Try giving it a `limit` in the query string. 

[1] https://github.com/quay/quay/blob/master/endpoints/v2/tag.py#L23

**Reply by inajob** [reply]
I set the `limit` parameter bug it doesn't work. So I read the reference  of Docker Registry API( https://docs.docker.com/registry/spec/api/#pagination-1 ) and set `n` parameter, it works untill 50.
This document says there is `last` parameter but it doesn't seem to work in quay.

**Reply by inajob** [reply]
Sorry I misunderstood this API, `last` require the ID of Link Header in previous response.
And it works!

My question is solved!

**Reply by ibazulic** [reply]
You can also use Quay's own API to list tags, you'll get more information about images. See my answer here for more info:

https://github.com/quay/quay/discussions/1807

You just add a parameter `page=n` at the end that signifies the tag page you want to list.
